### ISSUE TITLE
[Bug]Main Window Marked as Unresponsive when Dialog is open

### ISSUE THREAD

**Issue by 420noscope-exe** [guillaumechereau/goxel#379]
```
lsb_release -a                                                  
LSB Version:	n/a
Distributor ID:	ManjaroLinux
Description:	Manjaro Linux
Release:	24.0.4
Codename:	Wynsdey

uname -r                                                        
6.6.40-1-MANJARO
```
_You may see me use dialogue/dialog interchangably, dialog is shorthand._

Using action ci #405 AppImage.

Expected Behavior:
The main window should ignore all user inputs while file dialogs are open, but should still be "responsive" and not halt the program.  This would prevent accidental changes while the dialog is open, but would ensure that the program behaves.

Observed Behavior:
When opening a file dialog, the main window is marked "unresponsive" by the OS.  Attempting to move or resize the main window with a dialog open results in graphical artifacts. CPU usage of the "AppRun" process drops to 0% when a dialog is open.

Workaround:  Click "wait" or dismiss the message after the dialogue is closed.  Save frequently between imports.  Do not click Force Quit until you have saved.

Program becoming Unresponsive with open dialog:
![Screenshot from 2024-07-19 17-31-10](https://github.com/user-attachments/assets/0da746ef-59e5-4ddf-b3c0-3ed507e9a958)

Normal CPU utilization with no dialog open:
![Screenshot from 2024-07-19 17-37-21](https://github.com/user-attachments/assets/063dd212-f790-49f6-8566-662ce20303a8)

0% CPU utilization with open dialog:
![Screenshot from 2024-07-19 17-37-53](https://github.com/user-attachments/assets/a9b24726-0168-4507-bb62-2b13f0307ff1)

**Comment by guillaumechereau** [comment]
Yes that's an issue.  At least I think this should only affect the AppImage version now, since the build on linux uses gtk.
